client/src/components/Ratings&Reviews/AddReview.jsx

```diff
@@ -0,0 +1,479 @@
+import React from 'react';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+
+class AddReview extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.chooseStarRating = this.chooseStarRating.bind(this);
+    this.trackWordCount = this.trackWordCount.bind(this);
+    this.showPhotos = this.showPhotos.bind(this);
+    this.getProductName = this.getProductName.bind(this);
+    this.submitReview = this.submitReview.bind(this);
+    this.chooseCharacteristics = this.chooseCharacteristics.bind(this);
+    this.modalSummary = this.modalSummary.bind(this);
+    this.modalBody = this.modalBody.bind(this);
+    this.modalNickname = this.modalNickname.bind(this);
+    this.modalEmail = this.modalEmail.bind(this);
+    const { productId } = this.props;
+    this.state = {
+      overallStarChosen: 0,
+      newReview: {
+        product_id: productId, rating: 0, summary: '', body: '', recommend: '', name: '', email: '', photos: [], characteristics: {},
+      },
+      bodyWordCount: 0,
+      bodyWordCountCheck: false,
+      photos: [],
+      // urlPhotos: [],
+      productName: '',
+      // error checking
+      ratingError: false,
+      recommendError: false,
+      characteristicsError: false,
+      bodyError: false,
+      nicknameError: false,
+      emailError: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getProductName();
+  }
+
+  // GET product name
+  getProductName() {
+    const { productId } = this.props;
+    $.ajax({
+      url: `/atelier/products/${productId}`,
+      type: 'GET',
+      success: (data) => {
+        this.setState({
+          productName: data.name,
+        });
+      },
+    });
+  }
+
+  // Modal State Updates
+  // Functionality of choosing overall star rating in modal
+  chooseStarRating(e) {
+    const { newReview } = this.state;
+    const overallStar = e.target.id;
+    let num = Number(overallStar[overallStar.length - 1]);
+    let resetStars = 5;
+
+    this.setState({
+      overallStarChosen: Number(overallStar[overallStar.length - 1]),
```

Comment:
I think we could use num from line 63 here instead of re-pulling the star number?